### PR TITLE
Update the Charts Page

### LINKED ISSUE #431: Update the Charts Listing Page
### Scope

* Use new Material UI components and the Figma design files to update the Charts Listing Page

### PR DESCRIPTION
**Pull request by mikestone14**
Screenshot 2023-11-17 at 12 32 38 PM

Screenshot 2023-11-17 at 12 32 48 PM

closes #431

### CODE REVIEW

File: views/tables/ChartsTable/index.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        return (
+          <Link className="Link--unstyled" to={viewChart}>
+            <Typography variant="body1" sx={{ color: 'text.primary' }}>
+              {chart[property]}
+            </Typography>
+          </Link>
+        )
+      case 'updatedAt':
+        return chart[property]
+          ? moment(chart[property]).format(DATE_FORMAT)
```
Comment (Elkrival) [inline views/tables/ChartsTable/index.jsx:56]: I wonder if this is the time to use a different time library and slowly phase this out

File: views/tables/ChartsTable/TableMenu.jsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            <EditIcon sx={{ color: 'text.primary' }} />
+            <Typography sx={{ color: 'text.primary' }}>Edit</Typography>
+          </MenuItem>
+          <MenuItem
+            onClick={deleteAndClose}
+            disableRipple
+            disabled={!userOwnsChart}
+            sx={MENU_ITEM_STYLES}
+          >
+            <DeleteIcon />
```
Comment (Elkrival) [inline views/tables/ChartsTable/TableMenu.jsx]: I think the color is slightly greyish for this icon 
Screenshot 2023-11-17 at 12 41 48 PM

File: views/tables/ChartsTable/ChartsTable.test.jsx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+import { render, screen, within } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { routes } from '../../routes/routes'
+
+describe(ChartsTable, () => {
+  const user = createUser({ uid: 'user' })
+  const charts = [
+    createChart({
+      title: 'first chart',
+      updatedAt: '2023-06-28T15:43:23.789Z',
```
Comment (Elkrival) [inline views/tables/ChartsTable/ChartsTable.test.jsx:16]: what happens if we have a chart without this property

File: views/pages/ChartsPage.jsx [hunk 4]
```diff
@@ -79,12 +77,26 @@ const ChartsPage = () => {
...
-    <div>
-      <ChartList
-        handleShareChart={handleShareChart}
-        chartList={chartList}
-        removeChart={removeChart}
-        onDuplicateChart={onDuplicateChart}
+    <Box sx={{ p: '20px' }}>
+      <PageHeader
+        title="Charts"
+        cta={
```
Comment (Elkrival) [inline views/pages/ChartsPage.jsx:83]: what does cta stand for..? i just keep seeing Chicago Transit Authority

File: views/tables/ChartsTable/TableMenu.jsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+import MoreVertIcon from '@mui/icons-material/MoreVert'
+import EditIcon from '@mui/icons-material/Edit'
+import DeleteIcon from '@mui/icons-material/Delete'
+import ShareIcon from '@mui/icons-material/Share'
+import { Link } from 'react-router-dom'
+
+import { routes } from '../../routes/routes'
+import ChartModel from '../../models/ChartModel'
+
+const MENU_ITEM_STYLES = { display: 'flex', gap: '32px' }
```
Comment (Elkrival) [inline views/tables/ChartsTable/TableMenu.jsx:13]: should we put these in constants..?

**Review by Elkrival — COMMENTED** [review]
a few questions

File: views/pages/ChartsPage.jsx (hunk 4 above)
Comment (mikestone14) [reply views/pages/ChartsPage.jsx:83]: haha, CTA stands for Call To Action

File: views/tables/ChartsTable/TableMenu.jsx (hunk 5 above)
Comment (mikestone14) [reply views/tables/ChartsTable/TableMenu.jsx:13]: I don't think so, this is a constant for this component only.

File: views/tables/ChartsTable/ChartsTable.test.jsx (hunk 3 above)
Comment (mikestone14) [reply views/tables/ChartsTable/ChartsTable.test.jsx:16]: is that possible? I thought all charts had this.

File: views/tables/ChartsTable/index.jsx (hunk 1 above)
Comment (mikestone14) [reply views/tables/ChartsTable/index.jsx:56]: IMO we should have a separate stream of work to wrap moment out for something else. It's not used in a ton of places so wouldn't be too much of a burden.

File: views/tables/ChartsTable/TableMenu.jsx (hunk 2 above)
Comment (mikestone14) [reply views/tables/ChartsTable/TableMenu.jsx]: https://github.com/AMP-SCZ/dpdash/pull/482/commits/ac143e76b675ec551bf9979aeea593711a15d56d

File: views/tables/ChartsTable/ChartsTable.test.jsx (hunk 3 above)
Comment (Elkrival) [reply views/tables/ChartsTable/ChartsTable.test.jsx:16]: This property is missing and we it looks like there's an opportunity here to capture it

**Review by Elkrival — APPROVED** [review]
Looks good. I believe we need to set the updatedAt property to the model, but it's not a blocker

File: views/tables/ChartsTable/ChartsTable.test.jsx (hunk 3 above)
Comment (mikestone14) [reply views/tables/ChartsTable/ChartsTable.test.jsx:16]: we're already handling this property not being present, so I think we're covered 👍

**Comment by mikestone14** [comment]
> I believe we need to set the updatedAt property to the model, but it's not a blocker

In our controller we're setting this property when a chart is updated

**Pull request merged**